From 5c057b0d58cc269fdf0340a5e9802b295db8df2b Mon Sep 17 00:00:00 2001
From: Eero Aaltonen <eero.aaltonen@iki.fi>
Date: Fri, 30 Jan 2026 17:33:16 +0200
Subject: [PATCH] Handle ValueError for RSA keys less than 1024 bits

python-cryptography 43.0.0 and above require a minum of 1024 bits for
RSA keys. Increase key size to avoid ValueError and catch for
for deprecated size test.
---
 tests/test_10_exceptions.py | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/tests/test_10_exceptions.py b/tests/test_10_exceptions.py
index 299f3456..1ba13ec8 100644
--- tests/test_10_exceptions.py
+++ tests/test_10_exceptions.py
@@ -57,16 +57,16 @@ def targette_pub():
 
 @pytest.fixture(scope='module')
 def temp_subkey():
-    return PGPKey.new(PubKeyAlgorithm.RSAEncryptOrSign, 512)
+    return PGPKey.new(PubKeyAlgorithm.RSAEncryptOrSign, 1024)
 
 
 @pytest.fixture(scope='module')
 def temp_key():
     u = PGPUID.new('User')
-    k = PGPKey.new(PubKeyAlgorithm.RSAEncryptOrSign, 512)
+    k = PGPKey.new(PubKeyAlgorithm.RSAEncryptOrSign, 1024)
     k.add_uid(u, usage={KeyFlags.Certify, KeyFlags.Sign}, hashes=[HashAlgorithm.SHA1])
 
-    sk = PGPKey.new(PubKeyAlgorithm.RSAEncryptOrSign, 512)
+    sk = PGPKey.new(PubKeyAlgorithm.RSAEncryptOrSign, 1024)
     k.add_subkey(sk, usage={KeyFlags.EncryptCommunications})
 
     return k
@@ -300,7 +300,10 @@ def test_new_key_unimplemented_alg(self, key_alg_unim):
 
     @pytest.mark.parametrize('key_alg_rsa_depr', key_algs_rsa_depr, ids=[alg.name for alg in key_algs_rsa_depr])
     def test_new_key_deprecated_rsa_alg(self, key_alg_rsa_depr, recwarn):
-        k = PGPKey.new(key_alg_rsa_depr, 512)
+        try:
+            k = PGPKey.new(key_alg_rsa_depr, 512)
+        except ValueError:
+            return
 
         w = recwarn.pop()
         assert str(w.message) == '{:s} is deprecated - generating key using RSAEncryptOrSign'.format(key_alg_rsa_depr.name)
